docarray/document/abstract_document.py

```diff
@@ -1,7 +1,16 @@
-from typing import Dict, Iterable
+from abc import abstractmethod
+from typing import TYPE_CHECKING, Dict, Iterable, Type
 
 from pydantic.fields import ModelField
 
+if TYPE_CHECKING:
+    from docarray.document.mixins.proto import ProtoMixin
+
 
 class AbstractDocument(Iterable):
     __fields__: Dict[str, ModelField]
+
+    @classmethod
+    @abstractmethod
```

Comment:
this looks very weird. `AbstractDocument` caring about this. Why is this needed for every Mixin?

Reply:
I believe this is the exact same that having the __fields__ value at this level. This is a common behavior of all document to have this class and it is not only related to proto

Reply:
resolved now